### DISCUSSION TITLE
How to remove ssl certificate from website hosted in vercel

### DISCUSSION THREAD

**Discussion by Sourav9063** [vercel/community#2991] [Help]
### Summary

I made a website using ```NextJs``` that takes [list of websites](https://sourav9063.github.io/ftp-nextron/api/db.json) (that includes http:// and https:// websites) and checks whether the client can access the websites or not from client's network.

Here is the main [website](https://search-ftp.vercel.app/) and [source code](https://github.com/Sourav9063/search-ftp).

I hosted the website using [vercel](https://vercel.com/). Which provided me with an ssl certified website. But now I cannot fetch the ```http://*``` websites.
Getting this error:
```
Mixed Content: The page at '<URL>' was loaded over HTTPS, but requested an insecure resource '<URL>'. This request has been blocked; the content must be served over HTTPS.
```
[![enter image description here][1]][1]

Here is the check function:
```
export const getWorkingNotSureFromList = async ({
  setWorking,
  setNotSure,
  listOFLinks,
}) => {
  console.log("checking");
  listOFLinks.forEach(async (link) => {
    try {
      const response = await fetch(link, { mode: "no-cors" });

      if (response.ok) {
        console.log(link);
        setWorking((state) => {
          return [...state, { link: link, message: "Working" }];
        });
      } else {
        setWorking((state) => {
          return [...state, { link: link, message: "Working" }];
        });
      }
    } catch (e) {
      setNotSure((state) => {
        return [...state, { link: link, message: "Not Sure" }];
      });
    }
  });
};
```
**Note**
I cannot use a proxy server to overcome this error. Because the check has to be done in the client's ISP.

**Things I need help with** 

 - Is there any other way just to check whether the client has access to a website using JS in the browser overcoming CORS and Content Security Policy issues?Or,
 - Can I use ```window.open()``` and check if the url (not same origin) connects or not from my website?Or,
 - How to access ```http://*``` from ```https://*```.Or,
 - How can I allow insecure content for my website using js or meta tag?Or,
 - If there is no other way, How to remove ssl certificate provided by vercel.Or,
 - Is there any free hosting provider that hosts NextJs or static site that doesn't provide ssl certificate?(I know the security issue but my website doesn't require any.)

**Thanks for taking the time to read it.**

  [1]: https://i.stack.imgur.com/cfMMX.png

### Steps to Reproduce

Hosting in vercel will automatically provide ssl ceritificate.

**Comment by errorcodezero** [comment]
What if you proxied the site with cloudflare or another service and then had that not use an ssl certificate(for the request from client to cloudflare)

**Comment by smaeda-ks** [comment]
What you're trying to achieve is quite tricky, and there's no simple solution to this. Even if there's a way to host a website on HTTP on Vercel, the same issue is still there, because you're still making requests to HTTPS from HTTP page (and vice versa).

If you can't move the logic to server-side, the only way that I can possibly think of is hacky WebRTC approach

- https://incolumitas.com/2021/01/10/browser-based-port-scanning/
- https://medium.com/hownetworks/how-did-i-turn-my-browser-into-a-port-scanner-tricksy-but-doable-c37db85f9adc
